charms/worker/k8s/src/certificates.py

```diff
@@ -209,8 +258,13 @@ def _get_service_ips(self) -> Set[str]:
             ValueError: If the service CIDR is invalid.
         """
         service_ips = set()
-        service_cidrs = self._charm.config.get("bootstrap-service-cidr", "").split(",")
-        for cidr in service_cidrs:
+        service_cidrs = self._charm.config.get("bootstrap-service-cidr", "")
+        if not isinstance(service_cidrs, str):
+            log.warning("Service CIDRs is not str, instead %s", type(service_cidrs))
+            return set()
+        cidrs = service_cidrs.split(",")
+
+        for cidr in cidrs:
```

Comment:
if it's really just a runtime check for the type tester -- can we just use `cast` instead?
```suggestion
        service_cidrs = self._charm.config.get("bootstrap-service-cidr", "")
        cidrs = cast(str, service_cidrs).split(",")

        for cidr in cidrs:
```